src/main/java/de/allcom/services/ProductService.java

```diff
@@ -29,57 +30,49 @@ public class ProductService {
+    public ProductResponseValues saveProduct(SaveProductRequestDto saveProductRequestDto) {
 
-        Product savedProduct = productRepository.save(product);
+        if (saveProductRequestDto == null) {
+            throw new RestException(HttpStatus.NOT_FOUND, "SaveProductRequestDto is required");
+        }
+        Product product;
 
-        if (request.getImages() != null && !request.getImages().isEmpty()) {
-            List<ProductImage> photos = productImageService.uploadPhotos(request.getImages(), savedProduct);
-            savedProduct.setImages(photos);
+        if (saveProductRequestDto.getId() != null) {
+            product = productRepository.findById(saveProductRequestDto.getId())
+                    .orElseThrow(() -> new RestException(HttpStatus.NOT_FOUND,
+                            "Product with id: " + saveProductRequestDto.getId() + " did not found"));
         } else {
-            savedProduct.setImages(null);
+            product = new Product();
         }
+        Category category = categoryRepository.findById(saveProductRequestDto.getCategoryId())
+                .orElseThrow(() -> new RestException(HttpStatus.NOT_FOUND,
+                        "Category with id: " + saveProductRequestDto.getCategoryId() + " did not found"));
+
+        product.setName(saveProductRequestDto.getName());
+        product.setDescription(saveProductRequestDto.getDescription());
+        product.setWeight(saveProductRequestDto.getWeight());
+        product.setColor(saveProductRequestDto.getColor());
+        product.setCategory(category);
+        product.setState(Product.State.DRAFT);
+        product.setCreateAt(LocalDateTime.now());
+        product.setUpdateAt(LocalDateTime.now());
 
-        savedProduct = productRepository.save(savedProduct);
-
-        return converters.fromProductToProductDto(savedProduct);
-    }
+        product = productRepository.save(product);
+        List<MultipartFile> newImages = saveProductRequestDto.getImages();
 
-    @Transactional
-    public ProductDto updateProduct(UpdateProductRequestDto request, Long productId) {
-        Product product = productRepository.findById(productId).orElseThrow(() ->
-                new RestException(HttpStatus.NOT_FOUND, "The product is not found"));
-        if (product.getImages().isEmpty() && !request.getImageLinks().isEmpty()
-                || !product.getImages().isEmpty() && request.getImageLinks().isEmpty()) {
-            throw new RestException(HttpStatus.NOT_FOUND, "Links of photos did not found in the DB");
+        if (!newImages.isEmpty()) {
+            List<ProductImage> images = productImageService.uploadPhotos(newImages, product);
+            product.setImages(images);
+        }
+        if (product.getImages().isEmpty() && !saveProductRequestDto.getImageLinks().isEmpty()) {
+            throw new RestException(HttpStatus.NOT_FOUND,
+                    "Links of photos did not found in the DB");
```

Comment:
HttpStatus.BAD_REQUST, "In request "imageLinks" for product id < id> are not empty, but product hasn't imageLinks"